error: Could not compile `srml-staking`

macos

curl https://sh.rustup.rs -sSf | sh
rustup update nightly
rustup target add wasm32-unknown-unknown --toolchain nightly
rustup update stable
cargo install --git https://github.com/alexcrichton/wasm-gc
brew install cmake pkg-config openssl git llvm


Then
cargo install --force --git https://github.com/paritytech/substrate subkey

It showed that

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_RewardDestination::_parity_codec::Encode` is not satisfied
  --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/c93ca93/srml/staking/src/phragmen.rs:75:2
   |
75 |     /// Index of the candidate stored in the 'candidates' vector.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `_IMPL_DECODE_FOR_RewardDestination::_parity_codec::Encode` is not implemented for `usize`

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_RewardDestination::_parity_codec::Decode` is not satisfied
  --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/c93ca93/srml/staking/src/phragmen.rs:75:2
   |
75 |     /// Index of the candidate stored in the 'candidates' vector.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `_IMPL_DECODE_FOR_RewardDestination::_parity_codec::Decode` is not implemented for `usize`
   |
   = note: required by `_IMPL_DECODE_FOR_RewardDestination::_parity_codec::Decode::decode`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `srml-staking`.

Hi, it showed this error just now.

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_Call::_parity_codec::Encode` is not satisfied
   --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/4a14361/srml/aura/src/lib.rs:170:2
    |
170 |     start_slot: usize,
    |     ^^^^^^^^^^ the trait `_IMPL_DECODE_FOR_Call::_parity_codec::Encode` is not implemented for `usize`

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_Call::_parity_codec::Encode` is not satisfied
   --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/4a14361/srml/aura/src/lib.rs:172:2
    |
172 |     skipped: usize,
    |     ^^^^^^^ the trait `_IMPL_DECODE_FOR_Call::_parity_codec::Encode` is not implemented for `usize`

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_Call::_parity_codec::Decode` is not satisfied
   --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/4a14361/srml/aura/src/lib.rs:170:2
    |
170 |     start_slot: usize,
    |     ^^^^^^^^^^ the trait `_IMPL_DECODE_FOR_Call::_parity_codec::Decode` is not implemented for `usize`
    |
    = note: required by `_IMPL_DECODE_FOR_Call::_parity_codec::Decode::decode`

error[E0277]: the trait bound `usize: _IMPL_DECODE_FOR_Call::_parity_codec::Decode` is not satisfied
   --> /Users/zhangtong/.cargo/git/checkouts/substrate-7e08433d4c370a21/4a14361/srml/aura/src/lib.rs:172:2
    |
172 |     skipped: usize,
    |     ^^^^^^^ the trait `_IMPL_DECODE_FOR_Call::_parity_codec::Decode` is not implemented for `usize`
    |
    = note: required by `_IMPL_DECODE_FOR_Call::_parity_codec::Decode::decode`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `srml-aura`.

--- a/node/runtime/src/lib.rs
+++ b/node/runtime/src/lib.rs
@@ -59,7 +59,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_name: create_runtime_str!("substrate-node"),
 	authoring_version: 10,
 	spec_version: 89,
-	impl_version: 89,
+	impl_version: 90,
 	apis: RUNTIME_API_VERSIONS,
 };
 

--- a/srml/aura/src/lib.rs
+++ b/srml/aura/src/lib.rs
@@ -57,7 +57,6 @@ use primitives::traits::{SaturatedConversion, Saturating, Zero, One};
 use timestamp::OnTimestampSet;
 #[cfg(feature = "std")]
 use timestamp::TimestampInherentData;
-use parity_codec::{Encode, Decode};
 use inherents::{RuntimeString, InherentIdentifier, InherentData, ProvideInherent, MakeFatalError};
 #[cfg(feature = "std")]
 use inherents::{InherentDataProviders, ProvideInherentData};
@@ -163,7 +162,7 @@ decl_module! {
 }
 
 /// A report of skipped authorities in Aura.
-#[derive(Clone, Encode, Decode, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub struct AuraReport {
 	// The first skipped slot.

--- a/srml/aura/src/lib.rs
+++ b/srml/aura/src/lib.rs
@@ -51,7 +51,6 @@
 pub use timestamp;
 
 use rstd::{result, prelude::*};
-use parity_codec::Decode;
 use srml_support::storage::StorageValue;
 use srml_support::{decl_storage, decl_module};
 use primitives::traits::{SaturatedConversion, Saturating, Zero, One};
@@ -133,6 +132,7 @@ impl ProvideInherentData for InherentDataProvider {
 	}
 
 	fn error_to_string(&self, error: &[u8]) -> Option<String> {
+		use parity_codec::Decode;
 		RuntimeString::decode(&mut &error[..]).map(Into::into)
 	}
 }

--- a/srml/aura/src/lib.rs
+++ b/srml/aura/src/lib.rs
@@ -51,6 +51,7 @@
 pub use timestamp;
 
 use rstd::{result, prelude::*};
+use parity_codec::Decode;
 use srml_support::storage::StorageValue;
 use srml_support::{decl_storage, decl_module};
 use primitives::traits::{SaturatedConversion, Saturating, Zero, One};

